src/filters/Filters.js

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import _ from 'lodash';
+
+import {
+  Accordion,
+  FilterAccordionHeader,
+} from '@folio/stripes/components';
+
+import {
+  CheckboxFilter,
+  MultiSelectionFilter,
+} from '@folio/stripes/smart-components';
+
+import {
+  sources,
+  locations,
+  resourceTypes,
+  holdingStatuses,
+  languages,
+  filterNames,
+} from './model';
+
+const {
+  SOURCE,
+  LOCATION,
+  HOLDING_STATUS,
+  RESOURCE_TYPE,
+  LANGUAGE,
+} = filterNames;
+
+export default class Filters extends React.Component {
+  static propTypes = {
+    activeFilters: PropTypes.objectOf(PropTypes.array),
+    onChange: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    activeFilters: {},
+  }
+
+  createOnClearFilterHandler = (filterName) => () => {
+    this.props.onChange({
+      name: filterName,
+      values: [],
+    });
+  }
+
+  onChangeHandler = (filter) => {
+    this.props.onChange(filter);
+  }
+
+  isFilterNotEmpty(filterName) {
+    return _.get(this.props.activeFilters, [filterName, 'length']) > 0;
+  }
+
+  getLocationDataOptions() {
+    return locations.map((location) => ({
+      ...location,
+      disabled: this.isLocationFilterDisabled(),
+    }));
+  }
+
+  getHoldingStatusDataOptions() {
+    return holdingStatuses.map((holdingStatus) => ({
+      ...holdingStatus,
+      disabled: this.isHoldingStatusFilterDisabled(),
+    }));
+  }
+
+  getLanguageDataOptions() {
+    return languages.map((language) => ({
+      ...language,
+      disabled: this.isLanguageFilterDisabled(),
+    }));
+  }
+
+  isHoldingStatusFilterDisabled = () => {
+    const influencedFilter = this.props.activeFilters[SOURCE];
+    return influencedFilter && influencedFilter.includes('local');
```

Comment:
Please, remove hardcoded source values.
You can import sources from src/filters/model/index.js and filter the object 
I belive it will be buch more readable

Reply:
Have added constants